spec/serializers/current_user_serializer_spec.rb

```diff
@@ -352,4 +352,18 @@
       expect(initial_count).to eq(final_count)
     end
   end
+
+  describe "#featured_topic" do
+    fab!(:featured_topic) { Fabricate(:topic) }
+
+    before { user.user_profile.update(featured_topic_id: featured_topic.id) }
+
+    it "includes the featured topic" do
+      payload = serializer.as_json
+
+      expect(payload[:featured_topic]).to_not be_nil
+      expect(payload[:featured_topic][:id]).to eq(featured_topic.id)
+      expect(payload[:featured_topic][:title]).to eq(featured_topic.title)
```

Comment:
I don't think these assertions are sufficient since we are not really testing that the `BasicTopicSerializer` was used. Someone can accidentally switch this to another serializer and this test will still pass if the serializer has the `id` and `title` attribute. I think we should assert for the exact keys in the payload or maybe have some way to assert that the object was serialized with the `BasicTopicSerializer`.

Reply:
My intention was to trigger a BlockedSerializationError if we didn't use a serializer, but yes, you are right. I added another line to check if the result hash size is the same as the serializer.

Reply:
Maybe just check for the expected keys? That should be more reliable compared to the size.

Reply:
I mean, yes, but the keys are :id, :title, :fancy_title, :slug, :posts_count, which are indeed basic. I think all of the topic serializers inherit from BasicTopicSerializer, so having no serializer or any other topic serializer will still make those tests pass

Reply:
> so having no serializer or any other topic serializer will still make those tests pass

If we check for the exact keys, I believe the test will fail.